classes/local/datasource/images.php

```diff
@@ -140,4 +139,54 @@ public function config_on_update($data) {
             $this->cms->get('typeid')
         );
     }
+
+    /**
+     * Get configuration data for exporting.
+     *
+     * @return \stdClass|null
+     */
+    public function get_for_export(): ?\stdClass {
```

Comment:
null is returned by the default method defined in base. So if a datasource has nothing to export (like site), then it will not define the function and null will be returned by default.

Reply:
Changed